From 91cb5c93b9272ff5ccddff11019ba7d391176919 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 11 Dec 2024 14:44:52 +0800
Subject: [PATCH 275/317] AOSCOS: staging: sb105x: adapt to tty_struct changes

Per commit 6e94dbc7a4e4 ("tty: cumulate and document tty_struct::flow*
members"), struct members `stopped' and `tco_stopped' were grouped under
`struct flow.'.

Revise driver code accordingly.

Fixes: "AOSCOS: Revert "staging: sb105x: delete the driver""
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/staging/sb105x/sb_pci_mp.c   | 4 ++--
 drivers/staging/sb105x/sb_ser_core.h | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/staging/sb105x/sb_pci_mp.c b/drivers/staging/sb105x/sb_pci_mp.c
index 18e91bc61ba1..e333245ec9c5 100644
--- a/drivers/staging/sb105x/sb_pci_mp.c
+++ b/drivers/staging/sb105x/sb_pci_mp.c
@@ -480,7 +480,7 @@ static void __mp_start(struct tty_struct *tty)
 	struct sb_uart_port *port = state->port;
 
 	if (!uart_circ_empty(&state->info->xmit) && state->info->xmit.buf &&
-			!tty->stopped && !tty->hw_stopped)
+			!tty->flow.stopped && !tty->hw_stopped)
 		port->ops->start_tx(port);
 }
 
@@ -929,7 +929,7 @@ static int mp_get_lsr_info(struct sb_uart_state *state, unsigned int *value)
 
 	if (port->x_char ||
 			((uart_circ_chars_pending(&state->info->xmit) > 0) &&
-				!state->info->tty->stopped && !state->info->tty->hw_stopped))
+				!state->info->tty->flow.stopped && !state->info->tty->hw_stopped))
 		result &= ~TIOCSER_TEMT;
 
 	return put_user(result, value);
diff --git a/drivers/staging/sb105x/sb_ser_core.h b/drivers/staging/sb105x/sb_ser_core.h
index 8835415e1d3d..d8664e723b4a 100644
--- a/drivers/staging/sb105x/sb_ser_core.h
+++ b/drivers/staging/sb105x/sb_ser_core.h
@@ -52,7 +52,7 @@
 	(CIRC_SPACE((circ)->head, (circ)->tail, UART_XMIT_SIZE))
 
 #define uart_tx_stopped(port)		\
-	((port)->info->tty->stopped || (port)->info->tty->hw_stopped)
+	((port)->info->tty->flow.stopped || (port)->info->tty->hw_stopped)
 
 #define UART_ENABLE_MS(port,cflag)	((port)->flags & UPF_HARDPPS_CD || \
 					 (cflag) & CRTSCTS || \
-- 
2.48.1

